File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        int charsLength = chars.length;
        int res = 0;

        for (int i = 0; i < charsLength; i++) {
            Character current = chars[i];
            int currentCharLength = 1;

            while (i + 1 < charsLength && chars[i + 1] == current) {
                currentCharLength++;
                i++; 
            }

            chars[res++] = current;

            if (currentCharLength > 1) {
                for (char c : Integer.toString(currentCharLength).toCharArray()){
                    chars[res++] = c;
                }
            }
        }

        return res;
    }
}

File: 0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        idx, i = 0, 0

        while i < len(chars):
            ch = chars[i]
            count = 0

            while i < len(chars) and chars[i] == ch:
                i += 1
                count += 1
            
            chars[idx] = ch
            idx += 1

            if count > 1:
                for c in str(count):
                    chars[idx] = c
                    idx += 1
        
        return idx
            